tests/integration/preprocessor/_mask/test_mask.py

```diff
@@ -54,6 +54,30 @@ def setUp(self):
         self.mock_data = np.ma.empty((4, 3, 3))
         self.mock_data[:] = 10.
 
+    def test_components_fx_dict(self):
+        """Test compatibility of unput fx dictionary."""
+        iris.save(self.fx_mask, 'sftlf_test.nc')
```

Comment:
It would be good if you could save the file in a proper temporary directory, e.g. using the `tmp_path` fixture or [`tempfile.mkdtemp`](https://docs.python.org/3/library/tempfile.html#tempfile.mkdtemp).

Reply:
good call!

Reply:
ok - fixed that - it proves out that you can't call `pytests`'s `tmp_path` nor `tmp_dir` (which is old anyway) from a `tests.Test` class *and* you can't call a `python.fixture` directly *and* can't use it as functional argument inside the class. Bah! I was this close from actually changing the whole test module to separate pytest instances :grin:

Reply:
You're right, it's not easy to use fixtures with the unittest.Test. I found a way to make it work and pushed that, I hope you don't mind. Can you check if you think it's OK like this?

Reply:
just did and ran the test too, nice one @bouweandela - something a notch more advanced than my level of testing knowhow :beer:

Reply:
Got my inspiration from here: https://docs.pytest.org/en/latest/unittest.html#using-autouse-fixtures-and-accessing-other-fixtures ;-)